### ISSUE TITLE
`Segment.Item` does not have a focus ring

### ISSUE THREAD

**Issue by bryanforbes** [skeletonlabs/skeleton#3027]
### Current Behavior

Currently, a focused `Segment.Item` has no indication that it's focused:

<img width="495" alt="image" src="https://github.com/user-attachments/assets/29ce8a45-4964-4933-b2e4-91558b3caa2e">
<img width="670" alt="image" src="https://github.com/user-attachments/assets/4cb7d161-cef8-417c-b6b5-5fd13799eff3">

### Expected Behavior

A focused `Segment.Item` should have some sort of indication that it's focused, similar to text inputs

### Steps To Reproduce

1. Instantiate a `Segment` with several `Segment.Item` children
2. Tab to focus the control

### Link to Reproduction / Stackblitz

https://stackblitz.com/edit/sveltejs-kit-template-default-6b9mmo?file=src%2Froutes%2F%2Bpage.svelte

**Comment by endigo9740** [comment]
FYI, folding this issue under a single ticket so we can tackle this in one go:

- https://github.com/skeletonlabs/skeleton/issues/3030

**Comment by endigo9740** [comment]
@bryanforbes I'm going to reply here so you are notified.

After reviewing the Segment Control, I can con firm there *is* a focus state and focus styles. It's just not easy to see because the default styles for background/ring match (white in the Cerberus dark theme). If I change the background color for the active option, it then becomes more obvious that this is present:

https://github.com/user-attachments/assets/177df20b-3b50-40f8-99cd-a0ece2b1be0c

We use a global shared utility class called `focused` in Skeleton that provides the focus style in a uniform fashion across all components. The reason we do this, as opposed to outline, is because this is how Zag implements focus state. It's a conditional built-into the state machine itself. So we have to actively show/hide styles or classes to visually illustrate the state.

It's not exactly my preference for how focus should be handled in most scenarios, I always like to use browser native features for this sort of thing, but it does provide find grain control. I understand why the they do it. If you disagree with this approach, the best I can offer is to take this upstream to the Zag team.
